### PR TITLE
1.19

### PR DESCRIPTION
**Pull request by Bricktricker**
This PR ports this mod to Minecraft 1.19.2.

I moved the registration to DeferredRegister, fixed alle renames and converted the AddEntryLootModifier to use a Codec

I played a bit and I think everything is still working.

### CODE REVIEW

**Comment by KnightMiner** [comment]
> I moved the registration to DeferredRegister

As I asked in my previous PR, is this really nessesscary? I am not against deferred registers, but its a unneeded refactor.

**Comment by Bricktricker** [comment]
I couldn't find an alternative yesterday, but it looks like there is a [RegisterEvent](https://github.com/MinecraftForge/MinecraftForge/blob/1.19.x/src/main/java/net/minecraftforge/registries/RegisterEvent.java) that can be used
plus adding `@ObjectHolder` to every field. Should I use this?

**Comment by KnightMiner** [comment]
The code I had before works exactly the same in 1.19. Registry events were just changed so instead of being generic, the registry is part of a getter. Object holders are not nesseccary unless you expect someone to registry sub your items, and frankly no one should be doing registry subs on my modded items without talking to me first.

**Comment by Bricktricker** [comment]
Removed the DeferredRegister. I kept the diffrent registration methods (`registerEffects`, `registerBlocks`, ...) separate, but I can merge them into one method if you want to.

File: src/main/java/knightminer/simplytea/core/Registration.java [hunk 1]
```diff
@@ -62,22 +62,22 @@
...
 import net.minecraftforge.registries.ObjectHolder;
+import net.minecraftforge.registries.RegisterEvent;
+import net.minecraftforge.registries.RegisterEvent.RegisterHelper;
 
 import javax.annotation.Nonnull;
 import java.util.List;
 import java.util.Objects;
 
 @SuppressWarnings("unused")
-@ObjectHolder(SimplyTea.MOD_ID)
```
Comment (KnightMiner) [inline src/main/java/knightminer/simplytea/core/Registration.java:80]: I assume object holder no longer works at a class level, and must be applied to every field? If so, one of two fixes should be made:
* Apply object holder to every field and remove the injected hack
* Remove the injected hack

The injected hack was to stop IDEs from saying a non-null field was nullable as Forge makes a final null field non-null.

**Comment by KnightMiner** [comment]
Lets do a single event subscriber for registration, no sense having multiple subscribers to the same event in the same class.

**Comment by Bricktricker** [comment]
Merged all registration methods into one and removed the the injected hack.
The fields are now set in the registration method.

**Comment by KnightMiner** [comment]
I was unable to successfully build a workspace from this PR, were there gradle wrapper or build.gradle changes you missed in the PR?

**Comment by Bricktricker** [comment]
I commented out your mods.toml injection code in the build.gradle file, because it never worked for me. I just updated the values manually. Maybe thats the problem?

**Comment by KnightMiner** [comment]
That would prevent the game from launching, not the entire workspace from building.

You probably need to commit the required 1.19.2 changes to build.gradle (not commenting out the injection code, the other changes). You did not change that file in this PR, but in my experience it requires changes every MC release.

**Comment by Bricktricker** [comment]
I just ran `gradlew genEclipseRuns --refresh-dependencies` and it worked perfectly fine. I used this build.gradle file: https://gist.github.com/Bricktricker/b881f1531bdc8c8f5cd5f163986e2823

**Comment by KnightMiner** [comment]
Just answer my question directly. Did `build.gradle` have any changes **other than the stuff you commented out**?

**Comment by Bricktricker** [comment]
No, I haven't changed anything else.

**Comment by Mrbysco** [comment]
I was able to get this PR to work by adding `org.gradle.jvmargs=-Xmx3G` to the gradle.properties as it was running out of memory

**Pull request merged**